kotlinx-coroutines-core/common/src/flow/operators/Limit.kt

```diff
@@ -46,6 +46,7 @@ public fun <T> Flow<T>.dropWhile(predicate: suspend (T) -> Boolean): Flow<T> = f
 public fun <T> Flow<T>.take(count: Int): Flow<T> {
     require(count > 0) { "Requested element count $count should be positive" }
     return flow {
+        val ownershipMarker = Any()
```

Comment:
Micro-optimization: we definitely cannot use `this` flow instance, and we cannot use `FlowCollector` instance.

Maybe we can use something like `currentCoroutineContext` or the current continuation instance? 
Don't give it too much thought though, if it's not obvious, then probably it's not worth doing it here

Reply:
For me, it's not obvious. How does one obtain the current continuation instance without `suspendCoroutine` (which inpacts the state machine of the code)?